esp-hal/src/dma/pdma.rs

```diff
@@ -895,3 +883,17 @@ impl<'d> Dma<'d> {
         }
     }
 }
+
+impl<'d, C, M: Mode> Channel<'d, C, M>
+where
+    C: DmaChannel,
+{
+    /// Asserts that the channel is compatible with the given peripheral.
+    pub fn runtime_ensure_compatible(&self, peripheral: &PeripheralRef<'_, impl PeripheralMarker>) {
```

Comment:
Can this ever fail currently? I've been playing with it locally and I get a nice compile error if I pass a i2s dma channel into the spi driver on PDMA devices. Is this just forward thinking for the P4?

Reply:
Not today, but it'll be able to fail in the future. This is mostly intended for peripheral type erasure, mostly I2S on PDMA devices.